exercises/simple-linked-list/src/main/scala/SimpleLinkedList.scala

```diff
@@ -0,0 +1,10 @@
+import scala.reflect.ClassTag
+
+trait SimpleLinkedList[T] {
+  def isEmpty: Boolean
+  def value: T
+  def add(item: T): SimpleLinkedList[T]
+  def next: SimpleLinkedList[T]
+  def reverse: SimpleLinkedList[T]
+  def toArray(implicit m: ClassTag[T]): Array[T]
```

Comment:
Yes, I am aware of this. It is just that Scala's `Array` inherits all those bad things from Java including the need to use `ClassTag`.
But ok, if you think we should take the Readme literally then so be it. :)